### PR TITLE
Login and signup routes#7

### PR DESCRIPTION
**Pull request by giftofgrub**
Assuming you have the database running, you can test this out by:

1. creating a new user in "localhost:5000/users" with a post request, sending json (name, email, password)
2. logging in as that user via "localhost:5000/login" sending in email and password

### CODE REVIEW

File: server/app.py [hunk 1]
```diff
@@ -1,11 +1,112 @@
...
+#     # will return None
+#     current_user = get_jwt_identity()
+#     if current_user:
+#         return jsonify(logged_in_as=current_user), 200
+#     else:
+#         return jsonify(logged_in_as='anonymous user'), 200
+
+
+# if __name__ == '__main__':
+#     app.run()
```
Comment (skassam21) [inline server/app.py]: Remove any commented out code if you want this to go into dev - we usually don't ever push commented out code

Instead, you can make the code and add a comment as not used at the moment or something similar

File: server/models/User.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      nullable=False,
+      unique=True,
+  )
+
+  password = db.Column(
+      db.Text,
+      nullable=False,
+  )
+
+  def generate_user_object(self):
```
Comment (skassam21) [inline server/models/User.py]: I feel like to_dict or a method name like that is more appropriate

File: server/app.py [hunk 3]
```diff
@@ -1,11 +1,112 @@
...
+      access_token=create_access_token(identity=user_object)
+    ), 200
+  else:
+    # message = jsonify(message="Not authenticated. Please try again.")
+    # return Response(message, status=400, mimetype='application/json')
+    return jsonify(message="Not authenticated. Please try again."), 400
+
+
+# Chef: register as chef aroute
+@app.route('/register_chef')
```
Comment (skassam21) [inline server/app.py]: We don't need this route anymore right?

**Review by skassam21 — COMMENTED** [review]
Looks good, just a few minor comments

File: server/app.py (hunk 3 above)
Comment (giftofgrub) [reply server/app.py]: how about when a user decides to become a chef?

File: server/models/User.py (hunk 2 above)
Comment (giftofgrub) [reply server/models/User.py]: noted

File: server/app.py (hunk 3 above)
Comment (skassam21) [reply server/app.py]: i see that makes sense

Comment (giftofgrub) [reply server/app.py]: per this morning's discussion, we will not have a chefs table. Instead, we will have meals attached to users. I will take out this route then.

File: server/models/User.py [hunk 4]
```diff
@@ -1,16 +1,52 @@
...
+        try:
+            db.session.add(self)
+            db.session.commit()
+            return self
+        except IntegrityError:
+            db.session.rollback()
+            return None
+
+    
+    def authenticate(cls, email, password):
```
Comment (mssiegel) [inline server/models/User.py:40]: authenticate should have an @classmethod 
When made it into a classMethod I could create new users and login successfully. well done.

Comment (giftofgrub) [reply server/models/User.py:40]: thanks for finding this. yes, this should be a classmethod

**Review by mssiegel — APPROVED** [review]

**Pull request merged**